src/Concerns/FiresPublishingEvents.php

```diff
@@ -0,0 +1,311 @@
+<?php
+
+namespace Plank\Publisher\Concerns;
+
+use Illuminate\Database\Eloquent\Model;
+use Plank\Publisher\Contracts\PublishableEvents;
+
+/**
+ * @mixin Model
+ * @mixin PublishableEvents
+ */
+trait FiresPublishingEvents
+{
+    /**
+     * Persist the model was published even after the model is saved
+     */
+    public $wasPublished = false;
+
+    /**
+     * Persist the model was unpublished even after the model is saved
+     */
+    public $wasUnpublished = false;
+
+    /**
+     * Persist the model was drafted even after the model is saved
+     */
+    public $wasDrafted = false;
+
+    /**
+     * Persist the model was undrafted even after the model is saved
+     */
+    public $wasUndrafted = false;
+
+    public function initializeIsPublishable()
+    {
+        $this->addObservableEvents([
+            'beforePublishing',
+            'publishing',
+            'afterPublishing',
+            'beforeUnpublishing',
+            'unpublishing',
+            'afterUnpublishing',
+            'beforePublished',
+            'published',
+            'afterPublished',
+            'beforeUnpublished',
+            'unpublished',
+            'afterUnpublished',
+            'beforeDrafting',
+            'drafting',
+            'afterDrafting',
+            'beforeUndrafting',
+            'undrafting',
+            'afterUndrafting',
+            'beforeDrafted',
+            'drafted',
+            'afterDrafted',
+            'beforeUndrafted',
+            'undrafted',
+            'afterUndrafted',
+        ]);
+    }
+
+    public function fireBeforePublishing(): void
+    {
+        $this->fireModelEvent('beforePublishing');
+    }
+
+    public function firePublishing(): void
+    {
+        $this->fireModelEvent('publishing');
+        $this->wasPublished = true;
+    }
+
+    public function fireAfterPublishing(): void
+    {
+        $this->fireModelEvent('afterPublishing');
+    }
+
+    public function fireBeforeUnpublishing(): void
+    {
+        $this->fireModelEvent('beforeUnpublishing');
+    }
+
+    public function fireUnpublishing(): void
+    {
+        $this->fireModelEvent('unpublishing');
+        $this->wasUnpublished = true;
+    }
+
+    public function fireAfterUnpublishing(): void
+    {
+        $this->fireModelEvent('afterUnpublishing');
+    }
+
+    public function fireBeforePublished(): void
+    {
+        $this->fireModelEvent('beforePublished');
+    }
+
+    public function firePublished(): void
+    {
+        $this->fireModelEvent('published');
+        $this->wasPublished = false;
```

Comment:
Actually, this shold be moved to `afterPublished()`, the flag should only be set during the active model event cycle. So that if you save multiple times, you can check if it was published during that "save/create" etc.